x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsPersister.java

```diff
@@ -237,7 +237,7 @@ public void persistQuantiles(Quantiles quantiles) {
     public void persistQuantiles(Quantiles quantiles, WriteRequest.RefreshPolicy refreshPolicy, ActionListener<IndexResponse> listener) {
```

Comment:
This method can be called when reverting a model snapshot, and there's no guarantee that an autodetect process will have been started on the newer version of the product at the point when a model snapshot is reverted.  The call chain is `TransportRevertModelSnapshotAction.masterOperation()` -> `JobManager.revertSnapshot()` -> this method.  So one of those two calls needs to call `AnomalyDetectorsIndex.createStateIndexAndAliasIfNecessary()` first.